kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/status/NetworkStatusServiceAdapter.java

```diff
@@ -82,6 +90,68 @@ public Optional<NetworkInterfaceType> getNetInterfaceType(String ifName) {
         return ifType;
     }
 
+    public List<GwtWifiChannelFrequency> getAllSupportedChannels(String ifname) {
+        Optional<NetworkInterfaceStatus> netInterface = this.networkStatusService.getNetworkStatus(ifname);
+        if (!netInterface.isPresent() || !(netInterface.get() instanceof WifiInterfaceStatus)) {
+            return new ArrayList<>();
+        }
+
+        WifiInterfaceStatus wifiInterfaceInfo = (WifiInterfaceStatus) netInterface.get();
+        List<WifiChannel> channels = wifiInterfaceInfo.getChannels();
+
+        List<GwtWifiChannelFrequency> gwtChannels = new ArrayList<>();
+
+        for (WifiChannel channel : channels) {
+            gwtChannels.add(new GwtWifiChannelFrequency(channel.getChannel(), channel.getFrequency()));
+        }
```

Comment:
I expected to find here the channel filtering logic. As per our offline discussion we introduced #4425 to be able to filter channels in the UI as per our old algorithm:

```java
private boolean shouldBeDisabled(GwtWifiChannelFrequency channelFrequency) {
        boolean isNotValidCountryCode = this.countryCode.getValue().isEmpty()
                || this.countryCode.getValue().equals(WORLD_REGION_COUNTRY_CODE);
        return channelFrequency.isDisabled()
                || channelFrequency.isNoIrradiation() && !channelFrequency.isRadarDetection()
                || channelFrequency.isNoIrradiation() && channelFrequency.isRadarDetection() && isNotValidCountryCode;
    }
 ```

The `WifiChannel` data structure allows you to find this information.

Am I missing something?

Reply:
Forgot about that, adding it.